js/data/cryptoCurrencies.js

```diff
@@ -0,0 +1,163 @@
+import _ from 'underscore';
+import app from '../app';
+
+const currencies = [
+  {
+    name: 'Bitcoin',
+    code: 'BTC',
+    testnetCode: 'TBTC',
+    symbol: '₿',
+    baseUnit: 100000000,
+    minDisplayDecimals: 0,
+    maxDisplayDecimals: 8,
+    averageModeratedTransactionSize: 184,
+    feeBumpTransactionSize: 154,
+    qrCodeText: address => `bitcoin:${address}`,
+    icon: 'imgs/btcIcon128.png',
+    needCoinLink: 'https://openbazaar.org/bitcoin',
+    getBlockChainAddressUrl: (address, isTestnet) => (
+      isTestnet ?
+        `https://testnet.blockexplorer.com/address/${address}` :
+        `https://blockchain.info/address/${address}`
+    ),
+    getBlockChainTxUrl: (txid, isTestnet) => (
+      isTestnet ?
+        `https://testnet.blockexplorer.com/tx/${txid}` :
+        `https://blockchain.info/tx/${txid}`
+    ),
+  },
+];
+
+export default currencies;
+
+let _indexedCurrencies;
+
+function getIndexedCurrencies() {
+  if (_indexedCurrencies) return _indexedCurrencies;
+
+  _indexedCurrencies = currencies.reduce((indexedObj, currency) => {
+    indexedObj[currency.code] = _.omit(currency, 'code');
+    indexedObj[currency.testnetCode] = { ...currency };
+    return indexedObj;
+  }, {});
+
+  return _indexedCurrencies;
+}
+
+export function getCurrencyByCode(code) {
+  if (!code) {
+    throw new Error('Please provide a currency code.');
+  }
+
+  return getIndexedCurrencies()[code];
+}
+
+function getTranslatedCurrencies(lang = app.localSettings.standardizedTranslatedLang(),
+  sort = true) {
+  if (!lang) {
+    throw new Error('Please provide the language the translated currencies' +
+      ' should be returned in.');
+  }
+
+  let translated = currencies.map((currency) => ({
+    ...currency,
+    name: app.polyglot.t(`cryptoCurrencies.${currency.code}`),
+  }));
+
+  if (sort) {
+    translated = translated.sort((a, b) => a.name.localeCompare(b.name, lang));
+  }
+
+  return translated;
+}
+
+const memoizedGetTranslatedCurrencies =
+  _.memoize(getTranslatedCurrencies, (lang, sort) => `${lang}-${!!sort}`);
+
+export { memoizedGetTranslatedCurrencies as getTranslatedCurrencies };
+
+let currenciesSortedByCode;
+
+export function getCurrenciesSortedByCode() {
+  if (currenciesSortedByCode) {
+    return currenciesSortedByCode;
+  }
+
+  currenciesSortedByCode = currencies.sort((a, b) => {
+    if (a.code < b.code) return -1;
+    if (a.code > b.code) return 1;
+    return 0;
+  });
+
+  return currenciesSortedByCode;
+}
+
+/**
+ * Returns the currency data object based on the currency the connect to node is in.
+ */
+ // TODO - TODO - TODO: doc with TBTC vs BTC scenario.
+export function getServerCurrency() {
+  if (!app || !app.serverConfig || !app.serverConfig.cryptoCurrency) {
+    throw new Error('The cryptoCurrency field must be set on app.serverConfig.');
+  }
+
+  let curData = getCurrencyByCode(app.serverConfig.cryptoCurrency);
+
+  if (curData && curData.testnetCode !== app.serverConfig.cryptoCurrency) {
```

Comment:
Are you asking why I would preface the expression with `curData &&`? It's because `getCurrencyByCode` returns undefined if it doesn't find the currency.

(btw - I don't think you were reviewing the latest code. I don't see this line anymore in latest.)